### PR TITLE
Milestone 8: Views

### PR DESCRIPTION
**Pull request by Lizdev-05**
In this milestone, I achieved the following...

- Implemented the design from the sneak peek wireframes.
- Used the methods that I have created in the [Blog App - processing data in models](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/Processing_data_model_project.md).

### CODE REVIEW

File: app/views/users/index.html.erb [hunk 1]
```diff
@@ -1 +1,17 @@
...
\ No newline at end of file
+<h1>List of all users</h1>
+
+<table>
+<tr>
+  <th><b>Photo</b></th>
+  <th><b>Username</b></th>
+  <th><b>No of posts</b></th>
+</tr>
+<% @users.each do |user|%>
```
Comment (whiz25) [inline app/views/users/index.html.erb:9]: [OPTIONAL]
-You may create a _user.html file that renders the details of a user just like the one you have except without the loop. Then in the index.html.erb file all you have to do is `<%= render @users %>`

File: app/controllers/posts_controller.rb [hunk 2]
```diff
@@ -1,10 +1,10 @@
 class PostsController < ApplicationController
   def index
     @user = User.find(params[:user_id])
-    @posts = Post.where(user_id: params[@user.id])
+    @posts = @user.posts
```
Comment (whiz25) [inline app/controllers/posts_controller.rb:4]: - Please declare a method `recent_posts` to return the three most recent post for the user.

File: app/views/users/show.html.erb [hunk 3]
```diff
@@ -1 +1,16 @@
...
+<p><%= @user.bio %></p>
+
+<% @user.posts.each_with_index do |post, index|%>
+<p>Post #<%= index+1 %></p>
+<p>
+  <%= post.text%>
+</p>
+ <span>Likes : <%= post.likes_counter %></span>
+  <span>Comments : <%= post.comments_counter %></span>
+<% end %>
```
Comment (whiz25) [inline app/views/users/show.html.erb:15]: - Add a link to all user's posts. It's important for the user or others to have access to all posts of any user.

File: app/views/posts/index.html.erb [hunk 4]
```diff
@@ -1 +1,22 @@
...
+  <p>
+  <%= post.title %>
+</p>
+
+<p>
+<%= post.text %>
+</p>
+
+<ul>
+  <% post.comments.each do |comment| %>
```
Comment (whiz25) [inline app/views/posts/index.html.erb]: - Declare a method `recent_comments` to display the five most recent comments to a post.

**Review by whiz25 — CHANGES_REQUESTED** [review]
Hi @Lizdev-05, 

Good job so far! 
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights 🎉 🎉 🎉 
- Most requirements have been met
- Code looks clean and professional 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

### Martin Afani
#### Slack: whiz25

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/posts/index.html.erb (hunk 4 above)
Comment (Lizdev-05) [reply app/views/posts/index.html.erb]: Thank you for review.

**Review by Danieldotcomcoder — APPROVED** [review]
Hello @Lizdev-05 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it 👍 👍 😄 👏 

Congratulations! 🎉 🎉 🎉 

## Highlights
- Professional Readme File ✔✔
- Linters are setup correctly ✔✔
- Clear and nice code ✔✔

comp-2
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Lizdev-05** [comment]
Thank you for reviewing and approving my work

**Pull request merged**